src/frequenz/client/common/microgrid/__init__.py

```diff
@@ -1,4 +1,51 @@
 # License: MIT
-# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
 
-"""Frequenz microgrid definition."""
+"""Module to define the microgrid used with the common client."""
+
+from __future__ import annotations  # required for constructor type hinting
+
+from dataclasses import dataclass
+from typing import List, Self
+
+# pylint: disable=import-error, no-name-in-module
+from frequenz.api.common.v1.microgrid.microgrid_pb2 import (
+    MicrogridComponentIDs as PBMicrogridComponentIDs,
+)
+
+
+@dataclass(frozen=True, kw_only=True)
+class MicrogridComponentIDs:
+    """Linking component IDs with their respective microgrid ID."""
+
+    microgrid_id: int
+    """The ID of the microgrid."""
+
+    component_ids: List[int]
```

Comment:
Better use [`Sequence[int]`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence) here, as the class should be immutable. Otherwise people is not able to do `instance.component_ids = [1,2]` but they can still do `instance.component_ids.clear()` and `instance.component_ids.append(1)` for example.

```suggestion
    component_ids: Sequence[int]
```